### PR TITLE
Better eq simp

### PR DESCRIPTION
**Pull request by msooseth**
## Description
This does a bit better Eq IR simplification.

## Checklist

- [x] tested locally
- [ ] added automated tests
- [ ] updated the docs
- [x] updated the changelog

### CODE REVIEW

File: src/EVM/Expr.hs [hunk 1]
```diff
@@ -685,9 +685,14 @@ simplify e = if (mapExpr go e == e)
       | a == b = Lit 1
       | otherwise = Lit 0
     go (Eq (Lit 0) (Sub a b)) = Eq a b
-    go o@(Eq a b)
-      | a == b = Lit 1
-      | otherwise = o
+    go (Eq a b@(Lit _)) = Eq b a
```
Comment (d-xo) [inline src/EVM/Expr.hs:688]: This isn't covered by the `normArgs` stuff in Expr.hs?

File: src/EVM/Expr.hs [hunk 2]
```diff
@@ -685,9 +685,14 @@ simplify e = if (mapExpr go e == e)
...
+    go (Eq a b) =
+      case (aSimp, bSimp) of
```
Comment (d-xo) [inline src/EVM/Expr.hs:690]: Is this needed? Don't we already call `simplify` everywhere until fixpoint as part of `mapExpr`?

Comment (msooseth) [reply src/EVM/Expr.hs:690]: Oh of course, what was I thinking. Sorry.

**Comment by msooseth** [comment]
Honestly, this is not  a big gain. I think it's maybe worth it for the sanity of the produced equations. With this and the constant folding PR together, and `--quickcheck-replay 100` we get:

```
All 176 tests passed (296.21s)
*** Exception: ExitSuccess
```

Otherwise, we get:

```
All 176 tests passed (302.11s)
*** Exception: ExitSuccess
```

Basically, no difference (2% :laughing: ). So it doesn't do much. But it does make things more aligned, I think. Anyway, not important. I'll work on other stuff. I just spotted this and thought to fix it.

**Comment by msooseth** [comment]
Ah, one sec, there is more:

```
ghci> Expr.simplify (Max (Lit 0) (Lit 5))
Lit 0x5
ghci> Expr.simplify (Max (Lit 5) (Lit 0))
Max (Lit 0x5) (Lit 0x0)
```

Let me make this more thorough.

**Comment by msooseth** [comment]
OK, it should be better now... again, not sure this is the most important thing to do.

**Review by d-xo — APPROVED** [review]

**Pull request merged**